code/ship/ship.cpp

```diff
@@ -7458,6 +7476,13 @@ void ship_cleanup(int shipnum, int cleanup_mode)
 	object *objp = &Objects[shipp->objnum];
 	char *jumpnode_name = nullptr;
 
+	// Goober5000 - handle ship registry
+	auto entry = &Ship_registry[Ship_registry_map[shipp->ship_name]];
```

Comment:
An assert to check if the ship actually exists before this would be nice to catch potential issues more easily.

Reply:
:+1: